src/Symfony/Component/DomCrawler/Field/ChoiceFormField.php

```diff
@@ -45,11 +45,7 @@ class ChoiceFormField extends FormField
     public function hasValue()
     {
         // don't send a value for unchecked checkboxes
-        if (in_array($this->type, array('checkbox', 'radio')) && null === $this->value) {
-            return false;
-        }
-
-        return true;
+        return !in_array($this->type, array('checkbox', 'radio')) || null !== $this->value;
```

Comment:
IMO the original version was more readable.
Expression `<negation> || <negation>` is difficult to understand.

Reply:
I agree! Let's be careful with some of these changes. We must optimize code for humans, not machines. OPCache will compile this into the most optimized code possible.

Reply:
Gonna revert this one :+1: